### ISSUE TITLE
Remove CMB and add our custom code for meta fields

### ISSUE THREAD

**Issue by nikola3244** [WP-for-Church/Sermon-Manager#190]

**Comment by robertmain** [comment]
This is actually pretty good, because it looks like the entire Add/Edit interface is actually implemented so that it saved stuff as metadata. If we removed CMB, we could revert to using things like the default blog post content box that ships with wordpress (which would actually fix a part of #184!)

**Comment by nikola3244** [comment]
Yeah, interface is there, we just need to modify the engine. :wink: 

Unfortunately, I can't find time for it. :disappointed: 

If you have time, you are free and most welcome to start working on it - but beware, it won't be a small job.

**Comment by robertmain** [comment]
>If you have time, you are free and most welcome to start working on it 

I'm tempted

> but beware, it won't be a small job.

I'm not so sure...

1. The post content already goes into `post_content` in the `wp_posts` table. It also copied to `wp_postsmeta` under `sermon_description`. As I see it, we just need to re-enable the post body like I did for #196 which should put things back to how they were in that respect. Next after that is just to clean up the remaining `sermon_` metadata records, which a bit of SQL should do..

**Comment by nikola3244** [comment]
The reason why this is a big job is because this issue is about removing whole CMB library and replacing it with some custom engine. That custom engine should handle all exisiting meta fields rendering/sanitation/validation/escaping/saving. (most work would go into rendering, UI & UX work)

Also, sermon content is not yet saved into `post_content` (but that can be easily fixed), it's just in `sermon_description` meta. The content that is in `post_content` is our custom render of excerpt-like content. (`post_content` and `post_excerpt` are the same at the moment)

**Comment by nikola3244** [comment]
>>`post_content` and `post_excerpt` are the same at the moment
>
>Wait, what?

Okay, almost same. I've checked [the code](https://github.com/WP-for-Church/Sermon-Manager/blob/dev/sermons.php#L395) now. Excerpt is just shorter custom `post_content`

>Yeah, we need to get on that ASAP. I'm just wondering now if my assumptions about how this worked have messed anything up when I migrated my site with that SQL. I don't think I copied the post content into `sermon_description`.

Oh well, I think that we use `sermon_description` everywhere, but there might be fallbacks in place (I tend to overengineer things sometimes). It would be the best to copy everything to `sermon_description` as well

**Comment by robertmain** [comment]
I've actually rolled it back for the time being until I can figure out what's going on..

**Comment by nikola3244** [comment]
Backups are good :smile: 

Let me know if when you get any questions, I'll answer them

**Comment by robertmain** [comment]
I have questions related to sermon audio actually. I notice it saves `sermon_audio_id` and `sermon_audio`. Removing `sermon_audio_id` seems to have no effect, while `sermon_audio` seems to be where the actual mp3 URL is stored. Is `sermon_audio_id` actually used?

**Comment by nikola3244** [comment]
`sermon_audio_id` is not used by Sermon Manager anywhere, I'm not sure from where it got into your database

`sermon_audio` is the actual audio URL

PS: It would be very smart to change to using IDs in future, since URLs are hardcoded at the moment

**Comment by robertmain** [comment]
If I insert sermon audio, it generates `sermon_audio_id` as well as `sermon_audio`. Should I raise an issue for this?

**Comment by nikola3244** [comment]
Yeah, go for it. That's an unexpected behavior, and I notice couple entries in my test database as well - but oddly enough, searching through code for `sermon_audio_id` returned 0 results

**Comment by robertmain** [comment]
On second thoughts, searching the code for `sermon_audio_id` isn't showing anything for me either. Maybe I'll just leave it for now..
![image](https://user-images.githubusercontent.com/537684/39455932-286a68a4-4cb1-11e8-8a59-2e9b806a4f6f.png)
